src/compas_xr/project/assembly_extensions.py

```diff
@@ -0,0 +1,137 @@
+
+        Returns
+        -------
+        None
+
+        """
+        target_folder_path = os.path.join(folder_path, new_folder_name)
+        if not os.path.exists(target_folder_path):
+            os.makedirs(target_folder_path)
+
+        if z_to_y_remap:
+            frame = Frame(Point(0, 0, 0), Vector.Xaxis(), Vector.Zaxis())
+        else:
+            frame = Frame.worldXY()
+
+        for result in BrepGeometryConsumer(assembly).result:
+            brep = result.geometry
+            beam_frame = result.beam.frame
+            key = result.beam.key
+
+            brep_meshes = brep.to_meshes()
+            compas_mesh = Mesh()
+            for mesh in brep_meshes:
+                compas_mesh.join(mesh)
+            mesh_transformed = compas_mesh.transformed(Transformation.from_frame_to_frame(beam_frame, frame))
+
+            if not os.path.exists(target_folder_path):
+                raise Exception("File path does not exist {}".format(target_folder_path))
+            filename = "{}.obj".format(str(key))
+            mesh_transformed.to_obj(os.path.join(target_folder_path, filename))
+
+    def export_mesh_assembly_objs(self, assembly, folder_path, new_folder_name, z_to_y_remap=False):
+        """
+        Export Mesh assembly parts as .obj files to a folder path.
+
+        Parameters
+        ----------
+        assembly : compas.datastructures.Assembly
+            The Mesh (compas.datastructures.Mesh) assembly that you want to export parts from.
+        folder_path : str
+            The path in which you would like to create a storage folder.
+        new_folder_name : str
+            The name of the folder you would like to create.
+        z_to_y_remap : bool, optional
+            A boolean that determines if the z-axis should be remapped to the y-axis for .obj export. Default is False.
+
+        Returns
+        -------
+        None
+
+        """
+        target_folder_path = os.path.join(folder_path, new_folder_name)
+        if not os.path.exists(target_folder_path):
+            os.makedirs(target_folder_path)
+
+        if z_to_y_remap:
+            frame = Frame(Point(0, 0, 0), Vector.Xaxis(), Vector.Zaxis())
+        else:
+            frame = Frame.worldXY()
+
+        for part in assembly.parts():
+            # Mesh assembly can be made with or without a frame (ex. assembly.add_part(Mesh)) try & default to worldXY
+            if hasattr(part, "frame"):
+                part_frame = part.frame
+            else:
+                part_frame = Frame.worldXY()
+
+            # TODO: This is weird, but I can't transform a Part object, so I need to check if it's a Part or a Mesh
+            if isinstance(part, Part):
+                part_transformed = part.attributes["shape"].transformed(
+                    Transformation.from_frame_to_frame(part_frame, frame)
+                )
+            else:
+                part_transformed = part.transformed(Transformation.from_frame_to_frame(part_frame, frame))
```

Comment:
you can create the transform outside the `if` and use it
```suggestion
            transformation = Transformation.from_frame_to_frame(part_frame, frame)
            if isinstance(part, Part):
                part_transformed = part.attributes["shape"].transformed(transformation)
            else:
                part_transformed = part.transformed(transformation)
```

Reply:
Fixed in `TrajectoryElementAttachmentfromConfig` branch :)

Reply:
https://github.com/gramaziokohler/compas_xr/pull/7/commits/c284d4baec9fca8e88cf0ebde4a75e57a2b44f8e